### PR TITLE
Fix devcontainer features installation

### PR DESCRIPTION
**Pull request by mml5bg**
This PR implements native approach for dev container features installation.

### CODE REVIEW

File: .github/workflows/build-base-images.yml [hunk 1]
```diff
@@ -57,26 +56,12 @@ jobs:
...
-          file: ./Dockerfiles/${{ matrix.language }}/Dockerfile
-          platforms: linux/amd64, linux/arm64
-          tags: ${{ steps.meta.outputs.tags }}
-          labels: ${{ steps.meta.outputs.labels }}
\ No newline at end of file
+          imageName: ghcr.io/${{ github.repository }}/${{ matrix.language }}
+          imageTag: latest
+          push: always
+          platform: linux/amd64,linux/arm64
+          subFolder: ./Dockerfiles/${{ matrix.language }}/
```
Comment (kse3hi) [inline .github/workflows/build-base-images.yml]: Please fix all ```pre-commit run --all-files```  issues

File: Dockerfiles/cpp/.devcontainer/Dockerfile [hunk 2]
```diff
@@ -84,29 +82,35 @@ RUN apt-get --fix-missing update && apt-get upgrade -y && \
...
     ./install-velocitas-cli.sh ${TARGETARCH}
-USER 4000
+USER 1000
 RUN ./install-k3d-tooling.sh
 
-ADD --chown=4000 https://raw.githubusercontent.com/eclipse-velocitas/vehicle-app-cpp-sdk/main/requirements.txt  /container/requirements.txt
+ADD --chown=1000 https://raw.githubusercontent.com/eclipse-velocitas/vehicle-app-cpp-sdk/main/requirements.txt  /container/requirements.txt
 # Install Python requirements as vscode user and clean up
-RUN pip3 install -r /container/requirements.txt && sudo rm /container/requirements.txt
+RUN pip3 install -r /container/requirements.txt
```
Comment (kse3hi) [inline Dockerfiles/cpp/.devcontainer/Dockerfile:109]: You are installing all depenedncies on pip under "basecontainer" user. It means, that in runtime on vscode, all this dependencies will be not available for user "vscode".

**Review by kse3hi — CHANGES_REQUESTED** [review]

File: Dockerfiles/cpp/.devcontainer/Dockerfile (hunk 2 above)
Comment (doosuu) [reply Dockerfiles/cpp/.devcontainer/Dockerfile:109]: A few lines below the build dependencies from this base container are copied to `vscode` users user home, so they are available

File: .github/workflows/build-base-images.yml [hunk 3]
```diff
@@ -57,26 +56,12 @@ jobs:
...
         with:
-          push: true
-          context: "."
-          file: ./Dockerfiles/${{ matrix.language }}/Dockerfile
-          platforms: linux/amd64, linux/arm64
-          tags: ${{ steps.meta.outputs.tags }}
-          labels: ${{ steps.meta.outputs.labels }}
\ No newline at end of file
+          imageName: ghcr.io/${{ github.repository }}/${{ matrix.language }}
+          imageTag: latest
```
Comment (doosuu) [inline .github/workflows/build-base-images.yml]: Please add a step before this which changes the imageTag. If the branch this is executed from is 'main', 'latest' should be used, everything else should use the commit's sha value as the tag.

**Review by doosuu — CHANGES_REQUESTED** [review]

File: .github/workflows/build-base-images.yml (hunk 1 above)
Comment (doosuu) [reply .github/workflows/build-base-images.yml]: Added and will be pushed with the next commit

**Review by doosuu — APPROVED** [review]
LGTM

**Pull request merged**